### DISCUSSION TITLE
Help applying to my use case

### DISCUSSION THREAD

**Discussion by marcogallotta** [kymatio/kymatio#722] [General]
I am tryng to use kymatio's scattering in my project. I have a list `X` of 1D audio signals I want to classify using pytorch. Each signal is of variable length.

Firstly, I'm unsure whether my use case is considered 1D or 2D. I'd think 1D, because the signals are 1D, but since I have many of them is this considered 2D in your project? Otherwise, if it is actually 1D, I don't understand how to construct a scattering instance that can filter my *list* of signals.

Secondly, you req. a shape as input, but my signals are of variable input size. How do I handle this?

Lastly, with scattering, does the network design simply become the scattering object? Do you not need additional fully connected layers, etc.?

**Comment by OverLordGoldDragon** [comment]
1. **Variable length**: pad to same length.
  2. **Dim**: if they're independent channels, you can:
      - Group scattering coefficients per-channel by flattening: `(n_channels, n_coeffs, time)`. That's 2D.
      - Flatten completely: `(n_channels * n_coeffs, time)`. That's 1D. 
  3. **NN**: up to you, scattering coeffs can be fed directly to a linear classifier or be processed further by e.g. convolutions. The less data you have, the less learned layers are recommended.

Also be sure to normalize coefficients properly, they can be of vastly different energies across orders (zeroth, first, second). Can [log](https://www.kymat.io/gallery_1d/plot_classif_torch.html#log-scattering-transform) or [renormalize](https://raw.githubusercontent.com/scatnet/scatnet/master/doc/impl/impl.pdf) (sect 5).

**Reply by lostanlen** [reply]
@marcogallotta i will add that we have a long-term plan for analyzing signals of unbounded length, but it's not ready yet
